lib/git-fastclone.rb

```diff
@@ -343,17 +345,18 @@ def prefetch(submodule_file)
     # that this repo has been updated on this run of fastclone
     def store_updated_repo(url, mirror, repo_name, fail_hard)
       unless Dir.exist?(mirror)
-        Terrapin::CommandLine.new('git clone', '--mirror :url :mirror')
+        Terrapin::CommandLine.new('git clone', '--mirror :url :mirror' + NO_STDOUT_ERR)
```

Comment:
I think we still want at least the STDERR output to know why these actions failed. Ideally, we'd have them captured so that we could print them on failure. Otherwise, we'll have no context for why these actions are failing in the future.

Should we switch over to using the `build_execution` gem (we'd probably want to update OSS with our internal fork)? Or alternatively, we could use `open3` (or backticks if we're ok not outputting anything until the process completes) directly.

Reply:
I think it comes down the scope of this PR: 
1. assume switching to another dep (build_execution) is out of scope here, or not?
2. Add NO_STDOUT_ERR if no verbose mode? but then the bug is still there for verbose mode so probably no go
3. How about redirect to a log under `/private/var/tmp/git-fastclone/git_stdout/stderr.log` for example? Then we log saying to find those there? Guess the problem is we need this for each url and need to ensure we clear prior log... then might as well do option 1

Reply:
Replacing terrapin solves the bug, which is what I think of as the scope. I agree that replacing terrapin across all call sites in the file increases scope a bit, but it is confusing to have two different libraries handling shell commands from a maintenance perspective. If it were a significant lift, I'd think about it, but I'm assuming it is pretty close to a drop in replacement.

If we do go that route, we should look at what we get from terrapin to see if it is anything interesting that we don't get from build_execution.

Reply:
yeah going to try using the public build_execution version first so we can decide what to do with the internal version? in this case the major value of terrapin is to defend against bad input from user (rm -rf) but I feel that given the usage we don't have to? 
The other choice is to have both: then one can opt in/out to use build_exeuction with a flag? nah ppl can just stick to older gem version